src/sentry/static/sentry/app/components/events/relatedEvents/index.tsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import capitalize from 'lodash/capitalize';
+
+import space from 'app/styles/space';
+import {t} from 'app/locale';
+import DateTime from 'app/components/dateTime';
+import {Organization} from 'app/types';
+import EventView from 'app/utils/discover/eventView';
+import {transactionSummaryRouteWithQuery} from 'app/views/performance/transactionSummary/utils';
+import {PanelTable, PanelItem} from 'app/components/panels';
+import PlatformIcon from 'app/components/platformIcon';
+import Link from 'app/components/links/link';
+import {TableDataRow} from 'app/utils/discover/discoverQuery';
+import Tooltip from 'app/components/tooltip';
+import {generateEventSlug, eventDetailsRouteWithEventView} from 'app/utils/discover/urls';
+import {IconClock, IconFire, IconSpan} from 'app/icons';
+import TimeSince from 'app/components/timeSince';
+
+import {CURRENT_LOCATION} from './types';
+
+enum EVENT_TYPE {
+  ERROR = 'error',
+  TRANSACTION = 'transaction',
+}
+
+type Props = {
+  relatedEvents: Array<TableDataRow>;
+  eventView: EventView;
+  orgSlug: Organization['slug'];
+  currentLocation: CURRENT_LOCATION;
+};
+
+// List events that have the same tracing ID as the current Event
+const RelatedEvents = ({currentLocation, orgSlug, eventView, relatedEvents}: Props) => {
+  const getTransactionLink = (projectId: string, transationName: string) => {
+    return transactionSummaryRouteWithQuery({
+      orgSlug,
+      transaction: transationName,
+      projectID: projectId,
+      query: {
+        ...eventView.clone().getGlobalSelectionQuery(),
+        query: eventView.query,
+      },
+    });
+  };
+
+  const getEventTarget = (dataRow: TableDataRow & {type: EVENT_TYPE}) => {
+    if (currentLocation === CURRENT_LOCATION.DISCOVER) {
+      const eventSlug = generateEventSlug(dataRow);
+
+      return eventDetailsRouteWithEventView({
+        orgSlug,
+        eventSlug,
+        eventView,
+      });
+    }
+
+    return dataRow.type === EVENT_TYPE.ERROR
+      ? `/organizations/${orgSlug}/issues/${dataRow['issue.id']}/`
```

Comment:
It might be better to directly link to the related event for the related event's issue. Otherwise, the link will display the latest event which may not be the related event.

```suggestion
      ? `/organizations/${orgSlug}/issues/${dataRow['issue.id']}/events/${dataRow.id}/`
```